Jint/Native/JsValue.cs

```diff
@@ -142,6 +144,37 @@ public ArrayInstance AsArray()
             }
             return this as ArrayInstance;
         }
+        
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal IIterator GetIterator()
+        {
+            if (!(this is ObjectInstance oi))
+            {
+                ExceptionHelper.ThrowArgumentException("The value is not iterable");
+                return null;
+            }
+
+            // TODO
+            if (!oi.TryGetValue(GlobalSymbolRegistry.Iterator._value, out var value))
+            {
+                ExceptionHelper.ThrowArgumentException("The value is not iterable");
+                return null;
+            }
+
+            if (!(value is ICallable callable))
+            {
+                ExceptionHelper.ThrowArgumentException("The value is not iterable");
+                return null;
+            }
+
+            var obj = (ObjectInstance) callable.Call(this, System.Array.Empty<JsValue>());
```

Comment:
There is a `Arguments.Empty` field for this (same thing in the end, just more meaningful)